src/TolerantDefinitionResolver.php

```diff
@@ -0,0 +1,1177 @@
+        }
+        return null;
+    }
+
+    /**
+     * Gets Doc Block with resolved names for a Node
+     *
+     * @param Tolerant\Node $node
+     * @return DocBlock | null
+     */
+    private function getDocBlock(Tolerant\Node $node)
+    {
+        // TODO make more efficient (caching, ensure import table is in right format to begin with)
+        $docCommentText = $node->getDocCommentText();
+        if ($docCommentText !== null) {
+            list($namespaceImportTable,,) = $node->getImportTablesForCurrentScope();
+            foreach ($namespaceImportTable as $alias=>$name) {
+                $namespaceImportTable[$alias] = (string)$name;
+            }
+            $namespaceDefinition = $node->getNamespaceDefinition();
+            if ($namespaceDefinition !== null && $namespaceDefinition->name !== null) {
+                $namespaceName = (string)$namespaceDefinition->name->getNamespacedName();
+            } else {
+                $namespaceName = 'global';
+            }
+            $context = new Types\Context($namespaceName, $namespaceImportTable);
+
+            try {
+                return $this->docBlockFactory->create($docCommentText, $context);
+            } catch (\InvalidArgumentException $e) {
+                return null;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Create a Definition for a definition node
+     *
+     * @param Tolerant\Node $node
+     * @param string $fqn
+     * @return Definition
+     */
+    public function createDefinitionFromNode($node, string $fqn = null): Definition
+    {
+        $def = new Definition;
+        $def->fqn = $fqn;
+
+        // Determines whether the suggestion will show after "new"
+        $def->canBeInstantiated = $node instanceof Tolerant\Node\Statement\ClassDeclaration;
+
+        // Interfaces, classes, traits, namespaces, functions, and global const elements
+        $def->isGlobal = (
+            $node instanceof Tolerant\Node\Statement\InterfaceDeclaration ||
+            $node instanceof Tolerant\Node\Statement\ClassDeclaration ||
+            $node instanceof Tolerant\Node\Statement\TraitDeclaration ||
+
+            ($node instanceof Tolerant\Node\Statement\NamespaceDefinition && $node->name !== null) ||
+
+            $node instanceof Tolerant\Node\Statement\FunctionDeclaration ||
+
+            ($node instanceof Tolerant\Node\ConstElement && $node->parent->parent instanceof Tolerant\Node\Statement\ConstDeclaration)
+        );
+
+        // Static methods and static property declarations
+        $def->isStatic = (
+            ($node instanceof Tolerant\Node\MethodDeclaration && $node->isStatic()) ||
+
+            (($propertyDeclaration = TolerantParserHelpers::tryGetPropertyDeclaration($node)) !== null
+            && $propertyDeclaration->isStatic())
+        );
+
+        if ($node instanceof Tolerant\Node\Statement\ClassDeclaration &&
+            // TODO - this should be bette rrpreented in the parser API
```

Comment:
I think, that there should be a nicer way to get the base class of a class declaration?